sympy/matrices/tests/test_random.py

```diff
@@ -0,0 +1,463 @@
+import random
+
+from sympy import Matrix, eye, zeros, cartes, I
+from sympy.matrices.random import super_elementary_matrix
+from sympy.matrices.random import identity_matrix, jordan, jordan_normal, \
+    complex_to_real, invertible, regular_to_singular, diagonal_normal, \
+    diagonalizable, transposition, triangular, trigonalizable, \
+    isometry_normal, permutation, projection, elementary, rotation, \
+    reflection, normal, nilpotent, idempotent, singular, orthogonal, unitary,\
+    hermite, symmetric
+
+from sympy.matrices import random as _random
+from sympy.testing.pytest import raises
+
+TEST_DIMS = dict((d, tuple(cartes(range(d), range(d)))) for d in range(2, 6))
+TEST_PRECISION = 7
+
+random.seed(1)
+
+
+def _check_triangular(m):
+    s, t = m.shape
+    for i in range(s):
+        for j in range(i):
+            assert m[i, j] == 0, m
```

Comment:
The tests below would be clearer if this was just a boolean function e.g.:
```python
def is_triangular(m):
    s, t = m.shape
    return all(m[i, j] == 0 for i in range(s) for j in range(i))
```
Then it can be used like:
```python
assert is_triangular(M)
assert not is_triangular(N)
```

Reply:
Good point.